src/drivers/gitlab.js

```diff
@@ -176,23 +176,20 @@ class Gitlab {
     }
   }
 
-  async runnerByName(opts = {}) {
-    const { name } = opts;
-
+  async getRunners(opts = {}) {
     const endpoint = `/runners?per_page=100`;
     const runners = await this.request({ endpoint, method: 'GET' });
-    const runner = runners.filter(
-      (runner) => runner.name === name || runner.description === name
-    )[0];
-
-    if (runner) return { id: runner.id, name: runner.name };
-  }
-
-  async runnersByLabels(opts = {}) {
-    const { labels } = opts;
-    const endpoint = `/runners?per_page=100?tag_list=${labels}`;
-    const runners = await this.request({ endpoint, method: 'GET' });
-    return runners.map((runner) => ({ id: runner.id, name: runner.name }));
+    return await Promise.all(
+      runners.map(async ({ id, name, description, active, online }) => ({
+        id,
+        name: name || description,
```

Comment:
```suggestion
        name: description,
```

The `name` property is always `null` because [`POST /runners`](https://docs.gitlab.com/ee/api/runners.html#register-a-new-runner) doesn't accept it. We're using `description` instead of name to store a custom string.

Reply:
I would say this is true in the meantime GL fixes this bug on their side. The name should be changed and not the description so I would say is ok for now